controllers/invites.ts

```diff
@@ -0,0 +1,95 @@
+import { addInviteToInviteModel, getUserDiscordInvite } from "../models/discordactions";
+import { InviteBodyRequest } from "../types/invites";
+import {CustomResponse} from "../types/global";
+import { addLog } from "../models/logs";
+import { generateDiscordInviteLink } from "../utils/discord-actions";
+import { verifyAuthToken } from "../utils/verifyAuthToken";
+
+export const createInviteLink = async (req: InviteBodyRequest, res: CustomResponse) => {
+  try {
+    const { userId, purpose } = req.body;
+    const authHeader = req.headers?.authorization;
+    if (!authHeader) {
+      return res.boom.unauthorized();
+    }
+    const token = authHeader.split(" ")[1];
+    const isValid = await verifyAuthToken(token);
+    if (!isValid) {
+      return res.boom.unauthorized();
+    }
+
+    const inviteExist = await getUserDiscordInvite(userId);
+    if (!inviteExist.notFound) {
+      return res.boom.badRequest("Invite link already exists");
+    }
+
+    const inviteLink = await generateDiscordInviteLink();
+    if (!inviteLink) {
+      return res.boom.badRequest("Error while generating invite link");
+    }
+    const inviteData = {
+      userId,
+      purpose,
+      inviteLink,
+      createdAt: Date.now(),
+    };
+    const invite = await addInviteToInviteModel(inviteData);
+    if (!invite) {
+      return res.boom.badRequest("Error while adding invite link to database");
+    }
+    const inviteLog = {
+      type: "invite",
+      meta: {
+        action: "create",
+        createdBy: userId,
```

Comment:
isn't `createdBy` misleading? 
I can call this api with the appropriate token and pass anyone's userId right?

Reply:
> isn't `createdBy` misleading?
>
Then what? Can you suggest me another field name?

> I can call this api with the appropriate token and pass anyone's userId right?
>
Yes you can call this API with valid token any userId

Reply:
Maybe not have it all as the invite is not created by that userid. Instead its created for that userId.

Reply:
Adding createdBy for logs only

Reply:
The userId initiates a call to an external service API, which in turn invokes the RDS API to generate an invitation link. The link is created by the user, and the external service requires user involvement for the call; it does not occur automatically.

Reply:
Then what?